src/internal/kopia/base_finder.go

```diff
@@ -55,6 +55,20 @@ type BackupBase struct {
 	Reasons []identity.Reasoner
 }
 
+func (bb BackupBase) GetReasons() []identity.Reasoner {
+	return bb.Reasons
+}
+
+func (bb BackupBase) GetSnapshotID() manifest.ID {
+	return bb.ItemDataSnapshot.ID
+}
+
+func NewFromBackup(bup *backup.Backup) *BackupBase {
+	return &BackupBase{
+		Backup: bup,
+	}
+}
+
```

Comment:
I'm really uneasy about having a function like this when we also have a function like `GetSnapshotID` which explicitly tries to access unpopulated fields. It feels like too much of a gotcha. If this function is unused please remove it. Otherwise, we should come up with a way to make this safer

Reply:
Oh!  well, you're in luck.  This is dead code, anyway.  I just forgot to remove it.